examples/todo/src/main.rs

```diff
@@ -62,27 +63,32 @@ fn delete(id: i32, conn: db::Conn) -> Result<Flash<Redirect>, Template> {
 }
 
 #[get("/")]
-fn index(msg: Option<FlashMessage>, conn: db::Conn) -> Template {
+fn index(msg: Option<FlashMessage>, conn: db::SqliteDatabase) -> Template {
     Template::render("index", &match msg {
         Some(ref msg) => Context::raw(&conn, Some((msg.name(), msg.msg()))),
         None => Context::raw(&conn, None),
     })
 }
 
-fn rocket() -> (Rocket, Option<db::Conn>) {
-    let pool = db::init_pool();
-    let conn = if cfg!(test) {
-        Some(db::Conn(pool.get().expect("database connection for testing")))
-    } else {
-        None
-    };
-
+fn rocket() -> (Rocket, Option<db::SqliteDatabase>) {
     let rocket = rocket::ignite()
-        .manage(pool)
+        .attach(db::SqliteDatabase::fairing())
         .mount("/", routes![index, static_files::all])
         .mount("/todo/", routes![new, toggle, delete])
         .attach(Template::fairing());
 
+    let conn = if cfg!(test) {
```

Comment:
This is ugly and relies on types that aren't explicitly defined by the user. I would be happy for some recommendations here.

Reply:
You should implement (generate) a method on the attributed type that fetches a connection from the stored managed state. Something like:

```rust
/// Retrieves a connection of type `Self` from the `rocket` instance. Returns `Some` as long as
/// `Self::fairing()` has been attached and there is at least one connection in the pool.
fn get_one(rocket: &Rocket) -> Option<Self> {
    rocket.state::<Pool>()...
}
```

Then, this becomes:

```rust
let conn = match cfg!(test) {
    true => DbConn::get_one(&rocket),
    false => None
};
```

This makes it so that there are at least two methods that we're generating now. This probably means that we want a trait instead. Right now, it would be:

```rust
trait DatabaseConnection {
    trait Fairing: ::rocket::fairing::Fairing;

    fn fairing() -> Self::Fairing;
    fn get_one(&Rocket) -> Self;
}
```

Let's get everything working first and then move to a trait if it makes sense.